src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -111,24 +115,14 @@ public void Validate([NotNull] ModelValidationContext modelValidationContext)
             }
 
             // Check to avoid infinite recursion. This can happen with cycles in an object graph.
+            // Note that this is only applicble in case the model is pre-existing.
```

Comment:
consider the TryUpdateModel scenario. 
```c#
    public class SelfishPerson
    {
        public string Name { get; set; }
        public SelfishPerson MySelf { get { return this; } }
    }
```

If the model is like this and you call TryUpdateModel, we do not want to recurse for the property MySelf.